test/Faker/DefaultGeneratorTest.php

```diff
@@ -6,12 +6,18 @@
 
 final class DefaultGeneratorTest extends TestCase
 {
+    /**
+     * @group legacy
```

Comment:
As far as I can see, we do not run different groups of tests at the moment.

What is the benefit of adding this annotation, then?

Reply:
The symfony/phpunit-bridge has a super power, its default configuration (which we use) will fail the test if we use deprecated code... unless we mark the test as legacy.

Reply:
Sounds more like magic to me, to be honest!

Reply:
> Sounds more like magic to me, to be honest!

Maybe it is. ¯\_(ツ)_/¯ 

Here are some related docs: https://symfony.com/doc/current/components/phpunit_bridge.html#mark-tests-as-legacy

No matter if it is magic or not, it is an excellent feature. It can help us to be sure that we never use code that is deprecated. So, if our code is tested, a user may always be able to use Faker without triggering a deprecation notice. 

This will also help us to to know what code and tests we should remove before tagging 2.0

Reply:
@Nyholm 

I'm sure we would find out via static code analysis, so I think we do not need this.

Reply:
But since you added it and we can easily remove it with the [`general_phpdoc_annotation_remove`](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.17.2/doc/rules/phpdoc/general_phpdoc_annotation_remove.rst) fixer, I think we can keep it in for now.